packages/base/index.d.ts

```diff
@@ -50,14 +63,114 @@ export declare const utils: {
   /**
    * convert bigint to BigUInt64 little-endian hex string
    *
-   * @param num bigint or number type can using BigInt() to convert to bigint
+   * @param num
    */
-  toBigUInt64LE(num: bigint | string): string;
+  toBigUInt64LE(num: bigint): HexString;
 
   /**
    * convert BigUInt64 little-endian hex string to bigint
    *
    * @param hex BigUInt64 little-endian hex string
    */
-  readBigUInt64LE(hex: string): bigint;
+  readBigUInt64LE(hex: HexString): bigint;
+};
+
+export interface EpochSinceValue {
+  length: number;
+  index: number;
+  number: number;
+}
+
+export type SinceType = "epochNumber" | "blockNumber" | "blockTimestamp";
+
+export declare const since: {
+  /**
+   * Parse since and get relative or not, type, and value of since
+   *
+   * @param since
+   */
+  parseSince(
+    since: PackedSince
+  ):
+    | {
+        relative: boolean;
+        type: "epochNumber";
+        value: EpochSinceValue;
+      }
+    | {
+        relative: boolean;
+        type: "blockNumber" | "blockTimestamp";
+        value: bigint;
+      };
+
+  /**
+   * parse epoch from blockHeader.epoch
+   *
+   * @param epoch
+   */
+  parseEpoch(epoch: HexString): EpochSinceValue;
+
+  /**
+   * return larger one of two sinces
+   *
+   * @param one since in absolute-epoch-number format
+   * @param another since in absolute-epoch-number format
+   */
+  largerAbsoluteEpochSince(one: PackedSince, another: PackedSince): PackedSince;
```

Comment:
Change this to `maximumAbsoluteEpochSince`, then accept a [variable number of arguments](https://www.damirscorner.com/blog/posts/20180216-VariableNumberOfArgumentsInTypescript.html)

Reply:
updated